### PR TITLE
[docs]Fix wording of Many model training guidance 

### PR DESCRIPTION
**Pull request by Wendi-anyscale**
## Why are these changes needed?
Fix wording

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-overview/use-cases.rst [hunk 1]
```diff
@@ -99,8 +99,8 @@ How do I do many model training on Ray?
 There are three ways of using Ray to express this workload.
 
 1. If you have a large amount of data, use Ray Data (:ref:`Tutorial <mmt-datasets>`).
-2. If you have a small amount of data (<10GB), want to integrate with tools, such as wandb and mlflow, and you have less than 20,000 models, use Ray Tune (:ref:`Tutorial <mmt-tune>`).
-3. If your use case does not fit in any of the above categories, for example if you need to scale up to 1 million models, use Ray Core (:ref:`Tutorial <mmt-core>`), which gives you finer-grained control over the application. However, note that this is for advanced users and will require understanding of Ray Core :ref:`design patterns and anti-patterns <core-patterns>`.
+2. If you have a small amount of data (<10GB), want to integrate with tools, such as Weights&Biases, MLflow, and have less than 20,000 models, use Ray Tune (:ref:`Tutorial <mmt-tune>`).
```
Comment (richardliaw) [inline doc/source/ray-overview/use-cases.rst:102]:
```suggestion
2. If you have a small amount of data (<10GB), want to integrate with tools, such as Weights & Biases, MLflow, and have less than 20,000 models, use Ray Tune (:ref:`Tutorial <mmt-tune>`).
```

^ make sure you add the spaces around the ampersand

File: doc/source/ray-overview/use-cases.rst [hunk 2]
```diff
@@ -99,8 +99,8 @@ How do I do many model training on Ray?
...
+3. If your use case does not fit in any of the above categories(e.g., you need to scale up to 1 million models), use Ray Core (:ref:`Tutorial <mmt-core>`), which gives you finer-grained control over the application. This option requires you to understand Ray Core :ref:`design patterns and anti-patterns <core-patterns>`.
```
Comment (richardliaw) [inline doc/source/ray-overview/use-cases.rst:103]: usually, a space is expected before an opening parenthesis 

```suggestion
3. If your use case does not fit in any of the above categories (e.g., you need to scale up to 1 million models), use Ray Core (:ref:`Tutorial <mmt-core>`), which gives you finer-grained control over the application. This option requires you to understand Ray Core :ref:`design patterns and anti-patterns <core-patterns>`.
```

File: doc/source/ray-overview/use-cases.rst [hunk 3]
```diff
@@ -99,8 +99,8 @@ How do I do many model training on Ray?
...
 
 1. If you have a large amount of data, use Ray Data (:ref:`Tutorial <mmt-datasets>`).
-2. If you have a small amount of data (<10GB), want to integrate with tools, such as wandb and mlflow, and you have less than 20,000 models, use Ray Tune (:ref:`Tutorial <mmt-tune>`).
-3. If your use case does not fit in any of the above categories, for example if you need to scale up to 1 million models, use Ray Core (:ref:`Tutorial <mmt-core>`), which gives you finer-grained control over the application. However, note that this is for advanced users and will require understanding of Ray Core :ref:`design patterns and anti-patterns <core-patterns>`.
+2. If you have a small amount of data (<10GB), want to integrate with tools, such as Weights & Biases, MLflow, and have less than 20,000 models, use Ray Tune (:ref:`Tutorial <mmt-tune>`).
+3. If your use case does not fit in any of the above categories (e.g., you need to scale up to 1 million models), use Ray Core (:ref:`Tutorial <mmt-core>`), which gives you finer-grained control over the application. This option requires you to understand Ray Core :ref:`design patterns and anti-patterns <core-patterns>`.
```
Comment (cadedaniel) [inline doc/source/ray-overview/use-cases.rst:103]:
```suggestion
3. If your use case does not fit in any of the above categories (e.g., you need to scale up to 1 million models), use Ray Core (:ref:`Tutorial <mmt-core>`), which gives you finer-grained control over the application. This option requires you to understand the Ray Core :ref:`design patterns and anti-patterns <core-patterns>`.
```

File: doc/source/ray-overview/use-cases.rst [hunk 4]
```diff
@@ -99,8 +99,8 @@ How do I do many model training on Ray?
 There are three ways of using Ray to express this workload.
 
 1. If you have a large amount of data, use Ray Data (:ref:`Tutorial <mmt-datasets>`).
-2. If you have a small amount of data (<10GB), want to integrate with tools, such as wandb and mlflow, and you have less than 20,000 models, use Ray Tune (:ref:`Tutorial <mmt-tune>`).
-3. If your use case does not fit in any of the above categories, for example if you need to scale up to 1 million models, use Ray Core (:ref:`Tutorial <mmt-core>`), which gives you finer-grained control over the application. However, note that this is for advanced users and will require understanding of Ray Core :ref:`design patterns and anti-patterns <core-patterns>`.
+2. If you have a small amount of data (<10GB), want to integrate with tools, such as Weights & Biases, MLflow, and have less than 20,000 models, use Ray Tune (:ref:`Tutorial <mmt-tune>`).
```
Comment (cadedaniel) [inline doc/source/ray-overview/use-cases.rst:102]:
```suggestion
2. If you have a small amount of data (<10GB), want to integrate with tools such as Weights & Biases and MLflow, and have less than 20,000 models, use Ray Tune (:ref:`Tutorial <mmt-tune>`).
```

**Review by cadedaniel — APPROVED** [review]
grammar nits

**Review by richardliaw — APPROVED** [review]

**Pull request merged**